### PR TITLE
Enumerate darwin imports from chained fixups

### PR DESCRIPTION
**Pull request by mrmacete**
In latest arm64e executables there are no binding opcodes, while usually the same information can be reconstructed by walking the metadata indexed by the LC_DYLD_CHAINED_FIXUPS command and stored in linkedit.

The problem in doing so on images which have already been loaded in memory by dyld is that the actual chains are lost because the bitfields in the slots have already been replaced by the actual pointers.

This change reconstructs the correct slots by scanning the pages of the segments which are pointed to by the fixups metadata and looking for pointers to resolved imports.

### CODE REVIEW

File: gum/gumdarwinmodule.c [hunk 1]
```diff
@@ -30,6 +33,23 @@ typedef struct _GumEmitTermPointersContext GumEmitTermPointersContext;
 
 typedef struct _GumExportsTrieForeachContext GumExportsTrieForeachContext;
 
+typedef struct _GumChainedFixupsHeader GumChainedFixupsHeader;
+
+typedef struct _GumChainedStartsInImage GumChainedStartsInImage;
+typedef struct _GumChainedStartsInSegment GumChainedStartsInSegment;
+
+typedef uint32_t GumChainedImportFormat;
+typedef uint32_t GumChainedSymbolFormat;
```
Comment (oleavr) [inline gum/gumdarwinmodule.c:42]: For consistency we should be using GLib types here, as this code is also compiled on Windows.

File: gum/gumdarwinmodule.c [hunk 2]
```diff
@@ -99,6 +120,114 @@ struct _GumExportsTrieForeachContext
...
+  GUM_CHAINED_IMPORT          = 1,
+  GUM_CHAINED_IMPORT_ADDEND   = 2,
+  GUM_CHAINED_IMPORT_ADDEND64 = 3,
+};
+
+struct _GumChainedImport
+{
+  uint32_t lib_ordinal :  8,
+           weak_import :  1,
+           name_offset : 23;
```
Comment (oleavr) [inline gum/gumdarwinmodule.c:145]: This syntax may break MSVC compatibility, needs a closer look.

**Review by oleavr — CHANGES_REQUESTED** [review]
Thanks!! 🤘 Just a super-shallow round of review first as I'm deep down in a different context right now:

File: gum/gumdarwinmodule.c (hunk 2 above)
Comment (mrmacete) [reply gum/gumdarwinmodule.c:145]: aww i didn't think about windows at all. i may need some help / advice on this topic.

**Pull request merged**